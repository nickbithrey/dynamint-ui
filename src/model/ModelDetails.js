import React from 'react';

export default class ModelDetails extends React.Component {

	constructor(props) {
		super(props);
		this.state = {};
		this.updateField = this.updateField.bind(this);
		this.update = this.update.bind(this);
	}
	
	updateField(e) {
		e.preventDefault();
		const val = e.target.value;
		const name = e.target.getAttribute('data-item');
		this.setState({...this.state, [name]: val});
	}
	
	update(e) {
		e.preventDefault();
		this.props.update(this.state);
		this.props.history.push('/models');
	}
	
	render() {
		return (
			<form onSubmit={this.update}>
				<input type="text" name="reference" id="edit-reference" placeholder="Reference" data-item="reference" onChange={this.updateField} />
				<input type="text" name="description" id="edit-description" placeholder="Description" data-item="description" onChange={this.updateField} />
				<button type="submit">Create</button>
			</form>
		);
	}
}