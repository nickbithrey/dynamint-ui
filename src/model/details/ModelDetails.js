import React from 'react';

export default class ModelDetails extends React.Component {

	constructor(props) {
		super(props);
		this.updateField = this.updateField.bind(this);
		this.update = this.update.bind(this);
		this.back = this.back.bind(this);
		this.create = this.props.match.params.type === 'create';
		this.id = props.location.state.id;
	}
	
	componentDidMount() {
		if (!this.props.details) {
			this.props.load(this.id);
		}
	}
	
	updateField(e) {
		e.preventDefault();
		const val = e.target.value;
		const name = e.target.getAttribute('data-item');
		this.props.details[name]= val;
	}
	
	update(e) {
		e.preventDefault();
		if (this.create) {
			this.props.create(this.props.details);
		} else {
			this.props.update(this.id, this.props.details);
		}
		this.props.clear();
		this.props.history.goBack();
	}
	
	back(e) {
		e.preventDefault();
		this.props.clear();
		this.props.history.goBack();
	}
	
	render() {
		if (!this.props.details) {
			return <h1>Loading</h1>
		}
		return (
			<div>
				<form onSubmit={this.update}>
					reference: <input type="text" name="reference" id="edit-reference" placeholder="Reference" data-item="reference" onChange={this.updateField} defaultValue={this.props.details.reference} />
					description: <input type="text" name="description" id="edit-description" placeholder="Description" data-item="description" onChange={this.updateField} defaultValue={this.props.details.description} />
					<button type="submit">Create</button>
				</form>
				<button onClick={this.back}>Back</button>
			</div>
		);
	}
}