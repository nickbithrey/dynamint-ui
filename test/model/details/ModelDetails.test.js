import React from 'react';
import { shallow } from 'enzyme';

import ModelDetails from '~/model/details/ModelDetails';
import { Route, Link } from 'react-router-dom';

describe('<ModelDetails /> rendering', () => {

	it('renders standard component', () => {
		const load = jest.fn(() => []);
		const update = jest.fn(() => []);
		const create = jest.fn(() => []);
		const clear = jest.fn(() => []);
		const details = {
			reference: 'ref',
			description: 'desc'
		};
		const match = {
			params: {
				type: 'create'
			}
		};
		const location = {
			state: {}
		};
		const history = {
			goBack: jest.fn(() => [])
		};
		const wrapper = shallow( <ModelDetails 
				load={load} 
				update={update} 
				create={create} 
				clear={clear} 
				details={details}
				match={match}
				location={location}
				history={history}
		/> );
		
		expect(wrapper).toMatchSnapshot();
	});
});