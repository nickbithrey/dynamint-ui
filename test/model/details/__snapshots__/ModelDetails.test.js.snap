// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModelDetails /> rendering renders standard component 1`] = `
<div>
  <form>
    <Stack
      maxWidth={600}
    >
      <Field
        label="Reference"
        name="reference"
        update={[Function]}
        value="ref"
      />
      <Field
        label="Description"
        name="description"
        update={[Function]}
        value="desc"
      />
    </Stack>
    <Stack
      maxWidth={600}
    >
      <ModelAttribute
        attributes={Array []}
        updateAttributes={[Function]}
      />
    </Stack>
    <Button
      onClick={[Function]}
      text="Save"
    />
    <Button
      onClick={[Function]}
      text="Back"
    />
  </form>
</div>
`;
